app/controllers/users/phones_controller.rb

```diff
@@ -3,6 +3,7 @@ class PhonesController < ReauthnRequiredController
     include PhoneConfirmation
 
     before_action :confirm_two_factor_authenticated
+    before_action :redirect_if_not_phone_owner, only: :edit
```

Comment:
Added a check so that if the user is trying to modify a phone configuration that doesn't belong to them, they aren't taken to the edit screen (even though they can't modify it) but instead get routed to the account page.